packages/dd-trace/src/ci-visibility/dynamic-instrumentation/worker/index.js

```diff
@@ -0,0 +1,106 @@
+const { parentPort } = require('worker_threads')
+// TODO: move session to common place
+const session = require('../../../debugger/devtools_client/session')
```

Comment:
session should be simple to move

Reply:
on second thought, `session` should probably go together with `inspector_promises_polyfill.js` and `state.js` since they seem to be self contained? I'm honestly OK with importing code directly here, as this code is evolving quickly. I don't know if it makes sense to invest on this now

Reply:
thoughts @watson ?

Reply:
TL;DR: Leave `session.js` where it is for now.

This file currently acts a singleton, so it can't both be called from your code and from DI in the same worker thread. But I don't assume that's ever going to happen - just an observation.

If we were to move this to a generic shared location, there's a chance that others might use it for other things in the future. If so, this singleton behavior might throw someone for a loop. But since it's such simple code, I don't see any good reason to make this into a shared module. The `inspector_promises_polyfill.js` file on the other hand, should me moved to a central location.

So normally my conclusion to the above would be that we should each have our own version of `session.js` that just called a shared `inspector_promises_polyfill.js`. However, since the current `session.js` so used all over the place in my code, it would mean that all that would have to be refactored to allow a session object to be passed into the called functions. And I think that's too big of a change right now - hence my comment at the top about leaving it as-is for now.

Reply:
that makes sense, thanks for the feedback